olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/PrefetchPartitionsRequest.h

```diff
@@ -0,0 +1,164 @@
+   * @return A reference to the updated `PrefetchPartitionsRequest` instance.
+   */
+  inline PrefetchPartitionsRequest& WithPartitionIds(
+      PartitionIds partition_ids) {
+    partition_ids_ = std::move(partition_ids);
+    return *this;
+  }
+
+  /**
+   * @brief Gets the list of the partitions.
+   *
+   * @return The vector of strings that represent partitions.
+   */
+  inline const PartitionIds& GetPartitionIds() const { return partition_ids_; }
+
+  /**
+   * @brief Gets the billing tag to group billing records together.
+   *
+   * The billing tag is an optional free-form tag that is used for grouping
+   * billing records together. If supplied, it must be 4–16 characters
+   * long and contain only alphanumeric ASCII characters [A-Za-z0-9].
+   *
+   * @return The `BillingTag` string or `boost::none` if the billing tag is not
+   * set.
+   */
+  inline const boost::optional<std::string>& GetBillingTag() const {
+    return billing_tag_;
+  }
+
+  /**
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param billingTag The `BillingTag` string or `boost::none`.
+   *
+   * @return A reference to the updated `PrefetchTilesRequest` instance.
+   */
+  inline PrefetchPartitionsRequest& WithBillingTag(
+      boost::optional<std::string> billingTag) {
+    billing_tag_ = billingTag;
+    return *this;
+  }
+
+  /**
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param billingTag The rvalue reference to the `BillingTag` string or
+   * `boost::none`.
+   *
+   * @return A reference to the updated `PrefetchTilesRequest` instance.
+   */
+  inline PrefetchPartitionsRequest& WithBillingTag(std::string&& billingTag) {
+    billing_tag_ = std::move(billingTag);
+    return *this;
+  }
+
+  /**
+   * @brief Gets the request priority.
+   *
+   * The default priority is `Priority::LOW`.
+   *
+   * @return The request priority.
+   */
+  inline uint32_t GetPriority() const { return priority_; }
+
+  /**
+   * @brief Sets the priority of the prefetch request.
+   *
+   * @param priority The priority of the request.
+   *
+   * @return A reference to the updated `PrefetchPartitionsRequest` instance.
+   */
+  inline PrefetchPartitionsRequest& WithPriority(uint32_t priority) {
+    priority_ = priority;
+    return *this;
+  }
+
+  /**
+   * @brief Creates a readable format for the request.
+   *
+   * @param layer_id The ID of the layer that is used for the request.
+   *
+   * @return A string representation of the request.
+   */
+  std::string CreateKey(const std::string& layer_id,
+                        boost::optional<int64_t> version = boost::none) const {
+    std::stringstream out;
+    out << layer_id;
+    if (version) {
+      out << "@" << version.get();
+    }
+    if (GetBillingTag()) {
+      out << "$" << GetBillingTag().get();
+    }
+    return out.str();
```

Comment:
This doesn't say much. LayerID, version and billing tag is probably for every request in the same client similar. Please add something from the partitions, e.g. first and last partition ID.
Also i think priority should also be included in the key.